Makefile

```diff
@@ -0,0 +1,42 @@
+# StoryMachine Makefile
+# Self-documenting makefile - targets include usage descriptions
+
+.DEFAULT_GOAL := help
+.PHONY: help setup run clean build
+
+help: ## Display this help message
+	@echo "StoryMachine - Available targets:"
+	@echo
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "  \033[36m%-10s\033[0m %s\n", $$1, $$2}'
+
+setup: ## Install uv if not present and sync dependencies
+	@if ! command -v uv >/dev/null 2>&1; then \
+		echo "Installing uv..."; \
+		curl -LsSf https://astral.sh/uv/install.sh | sh; \
+		echo "Please restart your shell or run: source ~/.bashrc"; \
```

Comment:
An automatic `uv` installation setup without prompting can catch someone off guard (say if they want to install `uv` via their native package manager).

Also the instruction to source `.bashrc` is incorrect for macOS, which defaults to zsh.

Given that this is the only external dependency, I think we can leave this automation out entirely. The README makes it obvious that one should install `uv`, developers can figure this out.